File: react/src/types/mapElement.ts
import type { Dayjs } from 'dayjs';
import type { CircleOptions, PolylineOptions, PolygonOptions, MarkerOptions } from './mapOverlay';

export type MapElementCategory =
  | 'activeOutbreaks'
  | 'historicalOutbreaks'
  | 'wildbirdDeaths'
  | 'farms'
  | 'wildbirdMigrations';

export type MapElement = {
  category: MapElementCategory;
  minDate: Dayjs;
  maxDate: Dayjs;
  firstMarker: MarkerOptions | null;
  secondMarker?: MarkerOptions | null;
  circle: CircleOptions | null;
  polyline: PolylineOptions | null;
  polygons: Set<PolygonOptions> | null;
};

File: flutter/lib/types/map_element.dart
import 'package:google_maps_flutter/google_maps_flutter.dart';

enum MapElementCategory {
  currentPoultryOutbreaks,
  currentCattleOutbreaks,
  historicalOutbreaks,
  wildbirdDeaths,
  farms,
  wildbirdMigrations,
}

// MapElement is an intermediary object between backend response data and the MapOverlay for the purpose of front-end filtering without requesting a new response.
class MapElement {
  final MapElementCategory category;
  final DateTime minDate;
  final DateTime maxDate;
  final Marker? firstMarker;
  final Marker? secondMarker;
  final Circle? circle;
  final Polyline? polyline;
  final Set<Polygon>? polygons;

  MapElement({
    required this.category,
    required this.minDate,
    required this.maxDate,
    this.firstMarker,
    this.secondMarker,
    this.circle,
    this.polyline,
    this.polygons,
  });
}
